### ISSUE TITLE
PHP scraper results are inconsistent

### ISSUE THREAD

**Issue by EMorf** [catmanjan/anutimetable#176]
This has the effect of **not showing** all the lectures/tutes/labs etc. on the website. I did some digging around and it seems that the _**iid**_ field of [timetable.json](https://github.com/catmanjan/anutimetable/blob/4ea3a16ffb2615938b74c349f095cc16da1b534c/data/timetable.json) does not update correctly for some of the courses when the **PHP scraper** is used. I don't know PHP myself, but I assume this should be an easy fix.

**Comment by EMorf** [comment]
Can confirm this is also an issue with the Javascript parser... 😢

**Comment by samyex6** [comment]
Not quiet sure what you mean.. 
Does it only happen when you are trying to _update_ the json file? Because  if there're any changes when updating the source the iid would not match between and old & new version since I haven't added a difference check (need to find a reliable identifier to do this check).

**Comment by EMorf** [comment]
Sorry, I'll try to rephrase the issue and provide an example. The gist of it is that some courses do not appear correctly in the anutimetable.com website.

As an example, consider **ACST3001_S1**. According to _anutimetable.com_, this is the class schedule:
![e2](https://user-images.githubusercontent.com/8428597/34635885-2499bafc-f2ea-11e7-9bf8-3681cfaffd7e.png)

However, according to the official [ANU timetable](http://timetabling.anu.edu.au/sws2018/) there is one more lecture that is missing from our version:
![e1](https://user-images.githubusercontent.com/8428597/34635899-51f83244-f2ea-11e7-8aa3-776fb441330f.png)

The thing is, even though the extra lecture is present as an entry in timetable.json file, it does not appear in anutimetable.com:
![e3](https://user-images.githubusercontent.com/8428597/34635924-a6b1c66a-f2ea-11e7-9ea1-2b87f6b024ea.png)

This is an issue for most of the courses and I'm unsure what's causing it... From my digging around, it seems to occur when we have entries with the same iid in the _timetable.json_ file for the same course. In the example above, both lecture entries have an iid of 0 and only one of them is shown. I'm not entirely sure if this is the cause though as I'm not familiar with how the codebase works.

**Comment by catmanjan** [comment]
@samyex6 can you clarify what iid and lid are? I believe they are used for relationships regarding selecting incompatible classes

**Comment by catmanjan** [comment]
Hmm, does anyone have PHP available for test? I think the regex has become invalidated due to a formatting change

Can you try changing:

`/([a-zA-Z0-9]+)_.+?\s(.+)/` to `/([a-zA-Z0-9]+)_.+?-(.+)/`

I am not sure if the change has happened to all classes so we may have to check both?

**Comment by catmanjan** [comment]
@EMorf plz>?

**Comment by EMorf** [comment]
I'm not in canberra atm but I'll have a look when I get back on Thursday

**Comment by samyex6** [comment]
@catmanjan **iid** is used to link with the class label such as "Lecture01/ 001", "Tutorial01/ 002" etc, I wasn't sure what to call it so made it **iid** (info id) :^)
I'm in Sydney rn, will try to fix this (if it's still not being fixed) and add some comments for the scraper when I get back.

**Comment by EMorf** [comment]
@catmanjan @samyex6 the new regex seems to have fixed the issue. I have updated the scrapers and uploaded the new timetable. I also wrote a Python script to check for course activities with conflicting iids:

![image](https://user-images.githubusercontent.com/8428597/34828248-07bfa28e-f731-11e7-8a52-ebae67154a45.png)

Out of approx. **1200** courses, only **59** appear to be having issues but they are also inconsistent on the ANU website. For example, POPH8317_S1 has 5 lectures described as LectureA/01:

![image](https://user-images.githubusercontent.com/8428597/34828427-a42f3b20-f731-11e7-8d35-3fbb68710a87.png)

This seems wrong to me (should be LectureA/01, LectureB/01 etc., as is the case for most courses). But on our side, everything seems fine for now so crisis averted. 👍
